_posts/2021-01-12-magic-control.adoc

```diff
@@ -0,0 +1,281 @@
+---
+layout: post
+title: Bored with magic tricks?
+date: 2021-01-06 # Update the date before publication
+tags: vertx
+synopsis: How to control the amount of magic from Quarkus
+author: cescoffier
+---
+
+Just before my PTO, someone told me: '_I don't like magic._'
+In this context, _magic_ refers to the amount of hidden _stuff_ done by Quarkus under the hood for the sake of simplicity.
+It includes dependency injection, annotations, and so on.
+
+It's not the first time I get that kind of comment.
+And coming from the Vert.x project, it makes sense.
+Vert.x has almost no magic, and for a good reason: too much magic can be terrible and make production tuning utterly expensive.
```

Comment:
```suggestion
Vert.x has  no magic, and for a good reason: too much magic can be terrible and make production tuning utterly expensive.
```

Reply:
well... I kind of disagree. There is some magic in Vert.x. TCP round robin is a bit magic. Let's put _almost_ in parenthesis.

Reply:
Let's say TCP round robin on network ports to verticles is less magic than what Panache does to your entity 😉 

But parens are fine to me.